go/marshal/decode_test.go

```diff
@@ -456,6 +456,28 @@ func TestDecodeSlice(t *testing.T) {
 	assert.Equal([]string{"a", "b", "c"}, s)
 }
 
+func TestDecodeSliceEmpty(t *testing.T) {
+	assert := assert.New(t)
+	var s []string
+
+	err := Unmarshal(types.NewList(), &s)
+	assert.NoError(err)
+	assert.Equal([]string(nil), s)
```

Comment:
`assert.Nil(s)` would also work, but I like the way you have it, being explicit with the nil-ness here.